File: solutions/python/hamming/1/hamming.py
"""Hamming distance calculation."""


def distance(strand_a, strand_b):
    """Return the Hamming distance between equal-length strands."""
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")

    return sum(left != right for left, right in zip(strand_a, strand_b))

File: solutions/csharp/hamming/1/Hamming.cs
using Strand = string;

public static class Hamming
{
    public static int Distance(Strand first, Strand second)
    {
        if (first.Length != second.Length)
        {
            throw new ArgumentException("Strands must be of equal length.");
        }

        return first.Zip(second).Count(pair => pair.First != pair.Second);
    }
}
